Fabulous.XamarinForms/src/Fabulous.XamarinForms.Core/ViewUpdaters.fs

```diff
@@ -28,6 +28,120 @@ module ViewUpdaters =
+                    availableKeyedElements.[key] <- prevColl.[i]
+                | ValueNone -> rest.Add prevColl.[i]
+            
+            let contains (key:string voption) =
+                match key with
+                | ValueSome key -> availableKeyedElements.ContainsKey key
+                | ValueNone -> false
+                
+            let newKeyContains el =
+                match getKey el with
+                | ValueSome key -> newKeys.Contains key
+                | ValueNone -> false
+  
+            // Adjust the existing targetColl and create the new targetColl
+            for i in 0 .. coll.Length-1 do
+                let newChild = coll.[i]
+                let key = getKey newChild
+                let elems =
+                        seq{
+                            yield! rest
+                            yield! availableKeyedElements.Values
+                           }
+                if (prevColl|>Seq.exists (identical newChild)) then
+                    move i newChild
+                    match getKey newChild with
+                     | ValueSome key -> availableKeyedElements.Remove key |> ignore
+                     |_ ->rest.Remove newChild |> ignore
+                     
+                elif (key.IsSome)
+                then
+                  if(contains (key)) then 
+                     let previousKeyedElement = availableKeyedElements.[key.Value] // We should be safe here
+                     if(canReuse previousKeyedElement newChild) then
+                      update i previousKeyedElement newChild
+                      availableKeyedElements.Remove key.Value |> ignore
+                  else
+                     match (elems|>Seq.tryFind(fun c-> canReuse c newChild)) with
+                     | Some el ->
+                         update i el newChild
+                         match getKey el with
+                         | ValueSome key -> availableKeyedElements.Remove key |> ignore
+                         |_ ->rest.Remove el |> ignore
+                     | None ->
+                        create i newChild
+                else
+                   
+                    match (elems|>Seq.tryFind(fun c-> canReuse c newChild && (not <| newKeyContains c))) with
+                     | Some el ->
+                         update i el newChild
+                         match getKey el with
+                         | ValueSome key -> availableKeyedElements.Remove key |> ignore
+                         |_ ->rest.Remove el |> ignore
+                         
+                     | None ->
+                         create i newChild
+
+            let elems =
+                        seq{
+                            yield! rest
+                            yield! availableKeyedElements.Values
+                           }
+                        
+            for el in elems do
+                let index= prevColl|>Array.findIndex (fun c-> c=el)
+                remove index
```

Comment:
Same here. You could only remove elements still inside `rest`. All `availableKeyedElements` will have been reused at this point.

Reply:
Though on second thought, I think we need to change this part a bit.

Since we reused everything that was reusable and created the correct controls at the correct indexes, we are sure that for index 0 to index (coll.Length - 1) everything is correct.
What we really want here is to delete the eventual excess (from coll.Length to prevColl.Length - 1).

So we could only go with the previous while loop there was at the beginning of the PR.

But it makes it hard to unit test. In the tests, we want to know which elements did not make it (e.g. what's remaining in `rest` at the end).

So the only way I can think of handling this would be to change `remove: int -> unit` to `remove: (lastIndex: int) -> (nonReusedElements: ViewElement list)`.
`updateCollectionGeneric` would remove all items after the last index of `coll` (like we did before).
And the tests would check the removed elements like currently.

Reply:
Ok, i will try this

Reply:
Do i understand correclty, that we should delete all elements from targetColl that exceeds lastIndex?

Reply:
Yes